File: Python/QuickFind.py
__author__ = 'Sword'


class QuickFind:

    def __init__(self, int_list):
        self.int_list = int_list


    def union(self, int_list, p, q):
        pid = self.int_list[p]
        qid = self.int_list[q]
        for i in range(len(self.int_list)):
            if self.int_list[i]==pid:
                self.int_list[i]=qid

    def isConnected(self, p, q):
        return self.int_list[p]== self.int_list[q]

    def __str__(self):
        initial_list = ''
        for x in self.int_list:
            initial_list = initial_list + str(x)
        return initial_list





def main():
    my_list = [i for i in xrange(0,10)]
    pol = QuickFind(my_list)
    print pol
    pol.union(pol, 3, 5)
    pol.union(pol, 5, 6)
    pol.union(pol, 3, 1)
    pol.union(pol, 8, 0)
    pol.union(pol, 8, 2)
    pol.union(pol, 9, 3)

    print pol.isConnected(3,0)



if __name__ == "__main__":
    main()

File: src/QuickFind.java
/**
 * Created by deadlydragon00 on 7/15/2015.
 */
public class QuickFind {

    public static void main(String[] args) {
        QuickFind pol = new QuickFind(10);

        pol.union(3, 5);
        pol.union(5, 6);
        pol.union(3, 1);
        pol.union(8, 0);
        pol.union(8, 2);
        pol.union(9, 3);

        // check if 5 and 6 are connected
        System.out.println(pol.connected(5, 6));
    }


    private int[] id;

    /***
     * Initialize the elements
     *
     * @param n ist he number of element
     */
    QuickFind(int n) {
        id = new int[n];
        for (int i = 0; i < n; i++)
            id[i] = i;
    }

    /***
     * Connect both points p and q.
     *
     * @param p the first site
     * @param q the second site
     */
    void union(int p, int q) {
        int pid = id[p];
        int qid = id[q];
        for (int i = 0; i < id.length; i++)
            if (id[i] == pid)
                id[i] = qid;
    }

    /***
     * Test the connection of both site.
     *
     * @param p the first site
     * @param q the second site
     * @return true if both sites are connected otherwise false.
     */
    boolean connected(int p, int q) {
        return id[p] == id[q];
    }

}
